tables/checks.py

```diff
@@ -67,3 +68,21 @@ def max_word_count(element, *, max_words):
     if not isinstance(element, str):
         raise TypeError("Value must be a string")
     return len(element.split()) <= max_words
+
+
+@pa.extensions.register_check_method(check_type="element_wise")
+def postcode_list(element):
+    """Checks that a string can be split on commas and each element matches a basic UK postcode regex.
+
+    :param element: an element to check
+    :return: True if passes the check, else False
+    """
+    if not isinstance(element, str):
+        raise TypeError("Value must be a string")
+    postcode_regex = r"^[A-Z]{1,2}\d[A-Z\d]? ?\d[A-Z]{2}$"
```

Comment:
This is slightly different to the regex here https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/71b823014f2f02340da8c3271ee4cff69721c8a8/core/transformation/utils.py#L9

Should they be the same? Or use the same source?

Reply:
Existing postcode regex allows this postcode, which new regex marked as invalid: `SW1A 1AA1`. Based on the fact that this postcode is invalid, I will replace the existing POSTCODE_REGEX const with the new one and use this in `checks.py`.

<img width="654" alt="image" src="https://github.com/communitiesuk/funding-service-design-post-award-data-store/assets/48405095/74f3a489-5437-4bde-95fd-f0e055c105f8">

Reply:
Scratch that last message, changing `POSTCODE_REGEX` causes a bunch of tests to fail. I think best to revert to using existing regex regardless of whether or not it is objectively more accurate than the new one!